src/app/modules/item/pages/item-edit/item-edit.component.ts

```diff
@@ -115,29 +115,33 @@ export class ItemEditComponent implements OnDestroy, PendingChangesComponent {
     );
   }
 
-  private getItemChanges(): ItemChanges {
+  private getItemChanges(): ItemChanges | undefined {
     const urlFormControl = this.itemForm.get('url');
     const usesApiFormControl = this.itemForm.get('uses_api');
     const textIdFormControl = this.itemForm.get('text_id');
 
-    if (urlFormControl === null || usesApiFormControl === null || textIdFormControl === null) return {};
+    if (urlFormControl === null || usesApiFormControl === null || textIdFormControl === null) return undefined;
 
-    const itemFormValues = {
-      url: urlFormControl.value as string | null,
-      uses_api: usesApiFormControl.value as boolean,
-      text_id: textIdFormControl.value as string,
-    };
+    const itemFormValues: ItemChanges = {};
 
-    if (itemFormValues.url === '') itemFormValues.url = null;
+    const url = urlFormControl.value as string;
+    if (url !== this.initialFormData?.url) itemFormValues.url = url !== '' ? url : null;
+
+    const usesApi = usesApiFormControl.value as boolean;
+    if (usesApi !== this.initialFormData?.uses_api) itemFormValues.uses_api = usesApi;
+
+    const textId = textIdFormControl.value as string;
+    if (textId !== '') itemFormValues.text_id = textId;
 
     return itemFormValues;
   }
 
   private updateItem(): Observable<void> {
     return this.createChildren().pipe(
       switchMap(res => {
-        if (!this.initialFormData) return throwError(new Error('Invalid form'));
-        const changes: ItemChanges = this.getItemChanges();
+        if (!this.initialFormData) return throwError(new Error('Invalid initial data'));
+        const changes = this.getItemChanges();
+        if (!changes || !Object.keys(changes).length) return throwError(new Error('Invalid form'));
```

Comment:
🤨 
`changes` undefined means there is an error, right.
`changes` with no key just mean there is nothing to be changed, it is not an error.

Reply:
fixed